libp2p/connection.nim

```diff
@@ -130,8 +130,12 @@ proc readLp*(s: Connection): Future[seq[byte]] {.async, gcsafe.} =
       await s.readExactly(addr result[0], int(size))
   except LPStreamIncompleteError as exc:
     trace "remote connection ended unexpectedly", exc = exc.msg
+    result.setLen(0)
```

Comment:
so seriously, the easier way to _never_ have this class of bugs is to not use `result`, but rather stick to `return` and local named variables. setting length here is redundant/confusing, result will no longer escape implicitly on this code path with the raise in there, so why is it zeroed?

Reply:
Yes, agree with regards to result, I can see how it can become ambiguous and inconsistent.